go/parquet/internal/testutils/random.go

```diff
@@ -0,0 +1,383 @@
+	r := rand.New(rand.NewSource(seed))
+	for idx := range out {
+		out[idx] = randFloat64(r)
+	}
+}
+
+func randomByteArray(seed uint64, out []parquet.ByteArray, heap *memory.Buffer, minlen, maxlen int) {
+	heap.Resize(len(out) * (maxlen + arrow.Uint32SizeBytes))
+
+	buf := heap.Bytes()
+	r := rand.New(rand.NewSource(seed))
+	for idx := range out {
+		length := r.Intn(maxlen-minlen+1) + minlen
+		r.Read(buf[:length])
+		out[idx] = buf[:length]
+
+		buf = buf[length:]
+	}
+}
+
+func FillRandomByteArray(seed uint64, out []parquet.ByteArray, heap *memory.Buffer) {
+	const (
+		maxByteArrayLen = 12
+		minByteArrayLen = 2
+	)
+	randomByteArray(seed, out, heap, minByteArrayLen, maxByteArrayLen)
+}
+
+func FillRandomFixedByteArray(seed uint64, out []parquet.FixedLenByteArray, heap *memory.Buffer, size int) {
+	heap.Resize(len(out) * size)
+
+	buf := heap.Bytes()
+	r := rand.New(rand.NewSource(seed))
+	for idx := range out {
+		r.Read(buf[:size])
+		out[idx] = buf[:size]
+		buf = buf[size:]
+	}
+}
+
+func FillRandomBooleans(p float64, seed uint64, out []bool) {
+	dist := distuv.Bernoulli{P: p, Src: rand.NewSource(seed)}
+	for idx := range out {
+		out[idx] = dist.Rand() != float64(0.0)
+	}
+}
+
+func fillRandomIsValid(seed uint64, pctNull float64, out []bool) {
+	r := rand.New(rand.NewSource(seed))
+	for idx := range out {
+		out[idx] = r.Float64() > pctNull
+	}
+}
+
+func InitValues(values interface{}, heap *memory.Buffer) {
+	switch arr := values.(type) {
+	case []bool:
+		fillRandomIsValid(uint64(time.Now().Unix()), 1.0, arr)
+	case []int32:
+		FillRandomInt32(0, arr)
+	case []int64:
+		FillRandomInt64(0, arr)
+	case []float32:
+		FillRandomFloat32(0, arr)
+	case []float64:
+		FillRandomFloat64(0, arr)
+	case []parquet.Int96:
+		FillRandomInt96(0, arr)
+	case []parquet.ByteArray:
+		FillRandomByteArray(0, arr, heap)
+	case []parquet.FixedLenByteArray:
+		FillRandomFixedByteArray(0, arr, heap, 12)
+	}
+}
+
+func RandomByteArray(seed uint64, out []parquet.ByteArray, heap *memory.Buffer, minlen, maxlen int) {
+	heap.Resize(len(out) * (maxlen + arrow.Uint32SizeBytes))
+
+	buf := heap.Bytes()
+	r := rand.New(rand.NewSource(seed))
+	for idx := range out {
+		length := r.Intn(maxlen-minlen+1) + minlen
+		r.Read(buf[:length])
+		out[idx] = buf[:length]
+
+		buf = buf[length:]
+	}
+}
+
+// func RandomDecimals(n int64, seed uint64, precision int32) []byte {
+// 	r := rand.New(rand.NewSource(seed))
+// 	nreqBytes := pqarrow.DecimalSize(precision)
+// 	byteWidth := 32
+// 	if precision <= 38 {
+// 		byteWidth = 16
+// 	}
+
+// 	out := make([]byte, int(int64(byteWidth)*n))
+// 	for i := int64(0); i < n; i++ {
+// 		start := int(i) * byteWidth
+// 		r.Read(out[start : start+int(nreqBytes)])
+// 		// sign extend if the sign bit is set for the last generated byte
+// 		// 0b10000000 == 0x80 == 128
+// 		if out[start+int(nreqBytes)-1]&byte(0x80) != 0 {
+// 			memory.Set(out[start+int(nreqBytes):start+byteWidth], 0xFF)
+// 		}
+// 	}
+// 	return out
+// }
```

Comment:
this relied on the pqarrow package that is for the integration with parquet and arrow, so i commented it out for now until i push that chunk and it's tests.